src/Sarif.UnitTests/TestData/SarifCurrentToVersionOneVisitor/ExpectedOutputs/ResultLocations.sarif

```diff
@@ -101,14 +101,14 @@
         }
       ],
       "rules": {
-        "WEB1079.AttributeValueIsNotQuoted": {
+        "WEB1079": {
```

Comment:
>WEB1079 [](start = 9, length = 7)

When we were converting a dictionary of v2 rules to a dictionary of v1 rules, we could preserve the v2 keys in the v1 dictionary. Now that the `PreReleaseCompatibilityTransformer` is converting the dictionary to an array, the keys are lost before the v2-to-v1 transformer is invoked.

In fact, the v2 format has no place for this "friendly compound id". If you want a place for it, we could add it to v2 rule.

Reply:
Oh, right. These terms should be persisted to rule.name, can you go ahead and do that? still an open question whether we document using rule.id and rule.name in this way to compose a compound id (with an opaque stable and instable readable component in well known locations).

---
In reply to: [249572234](https://github.com/Microsoft/sarif-sdk/pull/1228#discussion_r249572234) [](ancestors = 249572234,249559914)

Reply:
That's a good idea and makes sense for this file. The problem is that the dictionary key is not _always_ a friendly name (consider `TST0001-2`) and there's no way to tell whether it was intended as a friendly name. Thoughts?

---
In reply to: [249572657](https://github.com/Microsoft/sarif-sdk/pull/1228#discussion_r249572657) [](ancestors = 249572657,249572234,249559914)